thetis/exner_eq.py

```diff
@@ -111,8 +111,18 @@ def residual(self, solution, solution_old, fields, fields_old, bnd_conditions=No
 
         # bnd_conditions are the shallow water bcs, any boundary for which
         # nothing is specified is assumed closed
+
         for bnd_marker in (bnd_conditions or []):
-            f += -self.test*(fac*qbx*self.n[0] + fac*qby*self.n[1])*self.ds(bnd_marker)
+            bc = bnd_conditions[bnd_marker]
+
+            keys = [*bc.keys()]
+            values = [*bc.values()]
+
+            for i in range(len(values)):
+                if float(values[i]) != 0.0:
```

Comment:
Hmm, this looks like a strange condition. Why 0 value is special? What if it is 1e-16? Maybe the "null" case could be handled with None or something?

Reply:
I specifically made this 0 because the only time I don't want the boundary to be considered is if the user has specified a closed boundary. If it's 1e-16 then there is some material that is leaving the system. I agree handling it with None might be better but it seemed like a lot of unnecessary extra code and boundary options for something that could be handled more simply.

Reply:
I think we can revisit the boundary conditions once bendy elements work